central/complianceoperator/v2/integration/service/service_impl.go

```diff
@@ -84,6 +90,22 @@ func (s *serviceImpl) ListComplianceIntegrations(ctx context.Context, req *v2.Ra
 		return nil, errors.Wrap(err, "failed to convert compliance integrations.")
 	}
 
+	// Enrich cluster status with sensor connection status.
+	for _, apiIntegration := range apiIntegrations {
+		cluster, found, err := s.clusterDS.GetCluster(ctx, apiIntegration.GetClusterId())
```

Comment:
[nit] Maybe we could change this to query the DataStore just once. Not sure if it's worth the extra code, I like how this reads now. WDYT?

Reply:
I like this idea! I'll adjust it.